<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EFXEditor</class>
 <widget class="QWidget" name="EFXEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>480</width>
    <height>646</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>EFX Editor</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_5">
   <item row="4" column="0" colspan="2">
    <widget class="QPushButton" name="m_testButton">
     <property name="toolTip">
      <string>See what the EFX does when it is run</string>
     </property>
     <property name="text">
      <string>&amp;Test Run</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="3" column="0" colspan="2">
    <widget class="QTabWidget" name="m_tab">
     <property name="toolTip">
      <string/>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="General">
      <attribute name="title">
       <string>General</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="2" column="2">
        <widget class="QToolButton" name="m_removeFixtureButton">
         <property name="icon">
          <iconset resource="qlcui.qrc">
           <normaloff>:/edit_remove.png</normaloff>:/edit_remove.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="5" column="2">
        <widget class="QToolButton" name="m_lowerFixtureButton">
         <property name="icon">
          <iconset resource="qlcui.qrc">
           <normaloff>:/down.png</normaloff>:/down.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="6" column="2" rowspan="2">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="2">
        <widget class="QToolButton" name="m_addFixtureButton">
         <property name="icon">
          <iconset resource="qlcui.qrc">
           <normaloff>:/edit_add.png</normaloff>:/edit_add.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="4" column="2">
        <widget class="QToolButton" name="m_raiseFixtureButton">
         <property name="icon">
          <iconset resource="qlcui.qrc">
           <normaloff>:/up.png</normaloff>:/up.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="7" column="0" colspan="2">
        <widget class="QGroupBox" name="m_orderGroup">
         <property name="title">
          <string>Fixture order</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QRadioButton" name="m_parallelRadio">
            <property name="toolTip">
             <string>All fixtures move in parallel</string>
            </property>
            <property name="text">
             <string>Parallel</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="m_serialRadio">
            <property name="toolTip">
             <string>The pattern propagates to each fixture in a sequential order</string>
            </property>
            <property name="text">
             <string>Serial</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="m_asymmetricRadio">
            <property name="toolTip">
             <string>Each fixture starts moving immediately with an offset</string>
            </property>
            <property name="text">
             <string>Asymmetric</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="0" rowspan="6" colspan="2">
        <widget class="QTreeWidget" name="m_tree">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>1</verstretch>
          </sizepolicy>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::ExtendedSelection</enum>
         </property>
         <property name="rootIsDecorated">
          <bool>false</bool>
         </property>
         <property name="itemsExpandable">
          <bool>false</bool>
         </property>
         <property name="allColumnsShowFocus">
          <bool>true</bool>
         </property>
         <attribute name="headerDefaultSectionSize">
          <number>85</number>
         </attribute>
         <column>
          <property name="text">
           <string>Step</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Fixture</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Reverse</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Start Offset</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Intensity</string>
          </property>
         </column>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="Movement">
      <attribute name="title">
       <string>Movement</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0" rowspan="2" colspan="2">
        <widget class="QFrame" name="m_previewFrame">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>1</verstretch>
          </sizepolicy>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Sunken</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QGroupBox" name="m_algorithmGroup">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Pattern</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QComboBox" name="m_algorithmCombo">
            <property name="toolTip">
             <string>Pattern for moving the mirror/head</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="2" rowspan="2">
        <widget class="QGroupBox" name="m_parametersGroup">
         <property name="title">
          <string>Parameters</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="12" column="1">
           <widget class="QSpinBox" name="m_yPhaseSpin">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>Lissajous pattern's Y phase</string>
            </property>
            <property name="maximum">
             <number>360</number>
            </property>
            <property name="value">
             <number>90</number>
            </property>
           </widget>
          </item>
          <item row="7" column="0" colspan="2">
           <widget class="Line" name="line_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="m_widthLabel">
            <property name="text">
             <string>Width</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="m_widthSpin">
            <property name="toolTip">
             <string>Value width of the pattern</string>
            </property>
            <property name="maximum">
             <number>127</number>
            </property>
            <property name="value">
             <number>127</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="m_heightLabel">
            <property name="text">
             <string>Height</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="m_heightSpin">
            <property name="toolTip">
             <string>Value height of the pattern</string>
            </property>
            <property name="maximum">
             <number>127</number>
            </property>
            <property name="value">
             <number>127</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="m_xOffsetLabel">
            <property name="text">
             <string>X offset</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="m_xOffsetSpin">
            <property name="toolTip">
             <string>Pattern's center point on the X axis</string>
            </property>
            <property name="maximum">
             <number>255</number>
            </property>
            <property name="value">
             <number>127</number>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="m_yOffsetLabel">
            <property name="text">
             <string>Y offset</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSpinBox" name="m_yOffsetSpin">
            <property name="toolTip">
             <string>Pattern's center point on the Y axis</string>
            </property>
            <property name="maximum">
             <number>255</number>
            </property>
            <property name="value">
             <number>127</number>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="m_rotationLabel">
            <property name="text">
             <string>Rotation</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QSpinBox" name="m_rotationSpin">
            <property name="toolTip">
             <string>Rotation of the pattern's starting point</string>
            </property>
            <property name="suffix">
             <string>°</string>
            </property>
            <property name="maximum">
             <number>359</number>
            </property>
           </widget>
          </item>
          <item row="9" column="0">
           <widget class="QLabel" name="m_xFrequencyLabel">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>X frequency</string>
            </property>
           </widget>
          </item>
          <item row="9" column="1">
           <widget class="QSpinBox" name="m_xFrequencySpin">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>Lissajous pattern's X frequency</string>
            </property>
            <property name="maximum">
             <number>32</number>
            </property>
            <property name="value">
             <number>2</number>
            </property>
           </widget>
          </item>
          <item row="10" column="0">
           <widget class="QLabel" name="m_yFrequencyLabel">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Y frequency</string>
            </property>
           </widget>
          </item>
          <item row="10" column="1">
           <widget class="QSpinBox" name="m_yFrequencySpin">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>Lissajous pattern's Y frequency</string>
            </property>
            <property name="maximum">
             <number>32</number>
            </property>
            <property name="value">
             <number>1</number>
            </property>
           </widget>
          </item>
          <item row="11" column="0">
           <widget class="QLabel" name="m_xPhaseLabel">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>X phase</string>
            </property>
           </widget>
          </item>
          <item row="13" column="0" colspan="2">
           <spacer name="verticalSpacer_4">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="11" column="1">
           <widget class="QSpinBox" name="m_xPhaseSpin">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>Lissajous pattern's X phase</string>
            </property>
            <property name="maximum">
             <number>360</number>
            </property>
            <property name="value">
             <number>90</number>
            </property>
           </widget>
          </item>
          <item row="12" column="0">
           <widget class="QLabel" name="m_yPhaseLabel">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Y phase</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QSpinBox" name="m_startOffsetSpin">
            <property name="suffix">
             <string>°</string>
            </property>
            <property name="maximum">
             <number>359</number>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="m_startOffsetLabel">
            <property name="text">
             <string>Start Offset</string>
            </property>
            <property name="textFormat">
             <enum>Qt::AutoText</enum>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QCheckBox" name="m_isRelativeCheckbox">
            <property name="text">
             <string>Relative</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QGroupBox" name="m_directionGroup">
         <property name="title">
          <string>Direction</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QRadioButton" name="m_forward">
            <property name="toolTip">
             <string>Run the pattern forwards</string>
            </property>
            <property name="text">
             <string>Forward</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="m_backward">
            <property name="toolTip">
             <string>Run the pattern backwards</string>
            </property>
            <property name="text">
             <string>Backward</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Minimum</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QGroupBox" name="m_runOrderGroup">
         <property name="title">
          <string>Run order</string>
         </property>
         <layout class="QVBoxLayout" name="_2">
          <item>
           <widget class="QRadioButton" name="m_loop">
            <property name="toolTip">
             <string>Run through over and over again</string>
            </property>
            <property name="text">
             <string>Loop</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="m_singleShot">
            <property name="toolTip">
             <string>Run through once and stop</string>
            </property>
            <property name="text">
             <string>Single shot</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="m_pingPong">
            <property name="toolTip">
             <string>First run forwards, then backwards, again forwards, etc...</string>
            </property>
            <property name="text">
             <string>Ping pong</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Minimum</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="1" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QToolButton" name="m_speedDial">
       <property name="toolTip">
        <string>Show/Hide speed dial window</string>
       </property>
       <property name="icon">
        <iconset resource="qlcui.qrc">
         <normaloff>:/speed.png</normaloff>:/speed.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="m_nameLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>EFX name</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="m_nameEdit">
       <property name="toolTip">
        <string>The name of the function being edited</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="qlcui.qrc"/>
  <include location="qlcui.qrc"/>
 </resources>
 <connections/>
</ui>
